Kickstarter-iOS/Features/PledgeOverTime/Views/Cells/PledgePaymentPlanCell.swift

```diff
@@ -0,0 +1,171 @@
+      self.rootStackView.trailingAnchor.constraint(equalTo: self.contentView.trailingAnchor),
+      self.rootStackView.topAnchor.constraint(equalTo: self.contentView.topAnchor),
+      self.rootStackView.bottomAnchor.constraint(equalTo: self.contentView.bottomAnchor)
+    ])
+
+    self.titleLabel.setContentCompressionResistancePriority(.required, for: .vertical)
+    self.titleLabel.setContentHuggingPriority(.required, for: .vertical)
+
+    self.subtitleLabel.setContentCompressionResistancePriority(.required, for: .vertical)
+    self.subtitleLabel.setContentHuggingPriority(.required, for: .vertical)
+
+    NSLayoutConstraint.activate([
+      self.checkmarkImageView.widthAnchor.constraint(equalToConstant: Styles.grid(4)),
+      self.checkmarkImageView.heightAnchor.constraint(equalTo: self.checkmarkImageView.widthAnchor)
+    ])
+
+    NSLayoutConstraint.activate([
+      self.checkmarkImageView.widthAnchor.constraint(equalToConstant: Styles.grid(4)),
+      self.checkmarkImageView.heightAnchor.constraint(equalTo: self.checkmarkImageView.widthAnchor)
+    ])
+  }
+
+  // MARK: - Styles
+
+  override func bindStyles() {
+    super.bindStyles()
+
+    self.selectionView.backgroundColor = .ksr_support_100
+
+    self.selectedBackgroundView = self.selectionView
+
+    applyRootStackViewStyle(self.rootStackView)
+
+    applyColumnStackViewStyle(self.leftColumnStackView)
+    self.leftColumnStackView.spacing = 0
+
+    applyColumnStackViewStyle(self.rigthColumnStackView)
+
+    applyTitleLabelStyle(self.titleLabel)
+
+    applySubtitleLabelStyle(self.subtitleLabel)
+
+    applyCheckmarkImageViewStyle(self.checkmarkImageView)
+  }
+
+  // MARK: - View model
+
+  override func bindViewModel() {
+    super.bindViewModel()
+
+    self.viewModel.outputs.checkmarkImageName
+      .observeForUI()
+      .observeValues { [weak self] imageName in
+        self?.checkmarkImageView.image = Library.image(named: imageName)
+      }
+
+    self.viewModel.outputs.titleText.observeForUI().observeValues { [weak self] titleText in
+      self?.titleLabel.text = titleText ?? ""
+    }
+
+    self.viewModel.outputs.subtitleText.observeForUI().observeValues { [weak self] subtitleText in
+      self?.configureSubtitleLabel(text: subtitleText)
+    }
+  }
+
+  func configureWith(value: PledgePaymentPlanCellData) {
+    self.viewModel.inputs.configureWith(data: value)
+  }
+
+  func configureSubtitleLabel(text: String?) {
+    guard let text = text, !text.isEmpty else {
+      self.rigthColumnStackView.removeArrangedSubview(self.subtitleLabel)
+      self.subtitleLabel.removeFromSuperview()
+      return
+    }
+
+    self.subtitleLabel.text = text
+    if !self.rigthColumnStackView.arrangedSubviews.contains(self.subtitleLabel) {
+      self.rigthColumnStackView.insertArrangedSubview(self.subtitleLabel, at: 1)
+    }
+  }
+}
+
+// MARK: - Styles helper
+
+private func applyRootStackViewStyle(_ stackView: UIStackView) {
```

Comment:
thanks for the suggestion! I checked this, and the function is not based on any existing prelude function or shared style helper in `Library/Styles`. Since it’s specific to this use case, I’ve kept it as is for now. Let me know if you’d like me to explore abstracting it further or adding a version to the shared styles for consistency.